c_binding/lsm_mgmt.cpp

```diff
@@ -2653,3 +2653,66 @@ int lsm_battery_list(lsm_connect *c, const char *search_key,
     rc = rpc(c, "batteries", parameters, response);
     return get_battery_array(c, rc, response, bs, count);
 }
+
+int lsm_volume_cache_info(lsm_connect *c, lsm_volume *volume,
+                          uint32_t *write_cache_policy,
+                          uint32_t *write_cache_status,
+                          uint32_t *read_cache_policy,
+                          uint32_t *read_cache_status,
+                          uint32_t *physical_disk_cache, lsm_flag flags)
+{
+    if (LSM_FLAG_UNUSED_CHECK(flags)) {
+        return LSM_ERR_INVALID_ARGUMENT;
+    }
+
+    int rc = LSM_ERR_OK;
+    CONN_SETUP(c);
+
+    if (!LSM_IS_VOL(volume)) {
+        rc = LSM_ERR_INVALID_ARGUMENT;
+        goto out;
+    }
+
+    if (write_cache_policy == NULL || write_cache_status == NULL ||
+        read_cache_policy == NULL || read_cache_status == NULL ||
+        physical_disk_cache == NULL) {
+
+        rc = LSM_ERR_INVALID_ARGUMENT;
+        goto out;
+    }
```

Comment:
If you enter the function and one or more of the input pointers are null then immediately return without touching any of the out variables.  Otherwise some are undefined and some are changed to unknown.  I see no issue with not changing an out parameter when the inputs are incorrect.  These are pointers which contain the address of an integer, I guess I don't consider them dangling if we don't change their value.